src/radical/pilot/pmgr/launching/base.py

```diff
@@ -25,34 +118,995 @@ def __init__(self, cfg, session):
+                             'action': rpc.TRANSFER})
+
+        # always stage the bootstrapper for each pilot, not in the tarball
+        # FIXME: this results in many staging ops for many pilots
+        bootstrapper_path = os.path.abspath("%s/agent/bootstrap_0.sh"
+                                           % self._root_dir)
+        pilot['sds'].append({'source': bootstrapper_path,
+                             'target': '%s/bootstrap_0.sh'
+                                     % pilot['pilot_sandbox'],
+                             'action': rpc.TRANSFER})
+
+        # always stage RU env helper
+        env_helper = ru.which('radical-utils-env.sh')
+        assert(env_helper)
+        self._log.debug('env %s -> %s', env_helper, pilot_sandbox)
+        pilot['sds'].append({'source': env_helper,
+                             'target': '%s/%s' % (pilot['pilot_sandbox'],
+                                                  os.path.basename(env_helper)),
+                             'action': rpc.TRANSFER})
+
+        # ----------------------------------------------------------------------
+        # we also touch the log and profile tarballs in the target pilot sandbox
+        pilot['fts'].append({
+                    'src': '/dev/null',
+                    'tgt': '%s/%s' % (pilot_sandbox, '%s.log.tgz' % pid),
+                    'rem': False})  # don't remove /dev/null
+        # only stage profiles if we profile
+        if self._prof.enabled:
+            pilot['fts'].append({
+                        'src': '/dev/null',
+                        'tgt': '%s/%s' % (pilot_sandbox, '%s.prof.tgz' % pid),
+                        'rem': False})  # don't remove /dev/null
+
+        # check if we have a sandbox cached for that resource.  If so, we have
+        # nothing to do.  Otherwise we create the sandbox and stage the RP
+        # stack etc.
+        #
+        # NOTE: this will race when multiple pilot launcher instances are used!
+        #
+        if resource not in self._sandboxes:
+
+            for sdist in sdist_paths:
+                base = os.path.basename(sdist)
+                pilot['fts'].append({'src': sdist,
+                                     'tgt': '%s/%s' % (session_sandbox, base),
+                                     'rem': False})
+
+            self._sandboxes[resource] = True
+
+        # ----------------------------------------------------------------------
+        # Create Job description and launch the pilot job
+
+        total_cpu_count = (requested_nodes * cores_per_node) or requested_cores
+        total_gpu_count = (requested_nodes * gpus_per_node)  or requested_gpus
+
+        jd_dict = ru.TypedDict()
+
+        jd_dict.name                  = job_name
+        jd_dict.executable            = '/bin/bash'
+        jd_dict.arguments             = shlex.split('-l ./bootstrap_0.sh %s'
+                                                                      % bs_args)
```

Comment:
@andre-merzky this breaks bootstrapping... revert back with `['-l ./bootstrap_0.sh %s' % bs_args]`?

Reply:
This is fixed now and handles spaced args (the split now happens on the `bs_args` and not on the full arg string).